test/test_mite_websockets.py

```diff
@@ -0,0 +1,63 @@
+from unittest.mock import patch
+
+import pytest
+from asyncmock import AsyncMock
+from mocks.mock_context import MockContext
+
+from websockets.exceptions import WebSocketException
+from mite_websocket import mite_websocket, WebsocketError
+
+
+@pytest.mark.asyncio
+async def test_mite_websocket_decorator():
+    context = MockContext()
+
+    @mite_websocket
+    async def dummy_journey(ctx):
+        assert ctx.websocket is not None
+
+    await dummy_journey(context)
+
+
+@pytest.mark.asyncio
+async def test_mite_websocket_decorator_uninstall():
+    context = MockContext()
+
+    @mite_websocket
+    async def dummy_journey(ctx):
+        pass
+
+    await dummy_journey(context)
+
+    assert getattr(context, "websocket", None) is None
+
+
+@pytest.mark.asyncio
+async def test_mite_websocket_connect_and_send():
+    context = MockContext()
+    url = "wss://foo.bar"
+    message = "baz"
+    connect_mock = AsyncMock()
+
+    @mite_websocket
+    async def dummy_journey(ctx):
+        await ctx.websocket.connect(url)
+        await ctx.websocket.send(message)
+
+    with patch("websockets.connect", new=connect_mock):
```

Comment:
Can you explain what is this testing?

Reply:
Sure, so the mocking stuff in Python is pretty powerful. What I'm saying here is that if at any point in this block, a call is made to websockets.connect, don't run that function, return a mock instead. My code will run and be tested as normal, but the underlying call to create the connection will never actually be made. 

The mocks returned by default can be called with any method and they'll be monkey patched on. So in this case once ctx.websocket.connection is set to a mock, I call my send method which calls the send method on the 'connection' which is actually a mock and itself returns another mock. Then I can check that both the connect and the send method were called as I expected. 

I then have to have some degree of faith the the websockets library works as desribed and that they've done their own testing.